### PR TITLE
MRG, ENH: Add include_tmax to cropping methods

### LINKED ISSUE #6652: `tmax` optionally exclusive in Epochs and cropping
Currently, `tmax` parameter in `Epochs` and `mne.epochs.BaseEpochs.crop` is inclusive. It would be convenient to be rather exclusive in some cases.
For example, in the example for "Decoding in time-frequency space data using the Common Spatial Pattern (CSP)" (https://mne-tools.github.io/stable/auto_examples/decoding/plot_decoding_csp_timefreq.html?highlight=csp), the grid search is used in the time domain by `crop` method:
```python
        # Crop data into time-window of interest
        X = epochs.copy().crop(w_tmin, w_tmax).get_data()
```
But this results in reusing one sample between two following windows!

What do you think? Is there any reason why it is strictly inclusive?

### PR DESCRIPTION
**Pull request by larsoner**
Closes #6652.

I stopped adding tests after doing Raw/Epochs/Evoked because they should be general enough to catch the others (SourceEstimate, AverageTFR, Dipole).

### CODE REVIEW

**Comment by larsoner** [comment]
cc @milan-rybar feel free to try this branch

File: mne/io/fiff/tests/test_raw_fiff.py [hunk 1]
```diff
@@ -968,6 +973,10 @@ def test_crop():
         raw1 = raw.copy().crop(tmin=tmin, tmax=tmin + 2)
         assert raw1[:][0].shape == (1, 2001)
 
+    # degenerate
+    with pytest.raises(ValueError, match='No samples.*when include_tmax=Fals'):
```
Comment (agramfort) [inline mne/io/fiff/tests/test_raw_fiff.py]:
```suggestion
    with pytest.raises(ValueError, match='No samples.*when include_tmax=False'):
```

Comment (larsoner) [reply mne/io/fiff/tests/test_raw_fiff.py]: Causes a PEP8 error (line length)

I usually just truncate them wherever necessary to get to 80 char

Comment (agramfort) [reply mne/io/fiff/tests/test_raw_fiff.py]: LOL. Feel free to merge then when CIs are green

**Comment by milan-rybar** [comment]
> cc @milan-rybar feel free to try this branch

Thank you for this

**Pull request merged**